prime-router/src/test/kotlin/ReportTests.kt

```diff
@@ -909,128 +933,288 @@ class ReportTests {
+                azuriteContainer.getMappedPort(
+                    10000
+                )
+            }/devstoreaccount1;QueueEndpoint=http://${azuriteContainer.host}:${
+                azuriteContainer.getMappedPort(
+                    10001
+                )
+            }/devstoreaccount1;"""
+        val blobContainerMetadata = BlobAccess.BlobContainerMetadata(
+            "container1",
+            blobConnectionString
+        )
+
+        mockkObject(BlobAccess.BlobContainerMetadata.Companion)
+        every { BlobAccess.BlobContainerMetadata.Companion.build(any(), any()) } returns blobContainerMetadata
+        every { BlobAccess.BlobContainerMetadata.Companion.build(any()) } returns blobContainerMetadata
+
+        val mockMetadata = mockk<Metadata> {
+            every { fileNameTemplates } returns emptyMap()
+        }
+        val mockActionHistory = mockk<ActionHistory> {
+            every { trackCreatedReport(any(), any(), blobInfo = any()) } returns Unit
+        }
+        val hl7MockData = UUID.randomUUID().toString().toByteArray() // Just some data
+        val receiver = Receiver(
+            "name", "org", Topic.FULL_ELR,
+            translation = Hl7Configuration(
+                receivingApplicationName = null, receivingApplicationOID = null,
+                receivingFacilityName = null, receivingFacilityOID = null, receivingOrganization = null,
+                messageProfileId = null
+            )
+        )
+
+        val reportIds = listOf(ReportId.randomUUID())
+        val (report, _, blobInfo) = Report.generateReportAndUploadBlob(
+            Event.EventAction.PROCESS, hl7MockData, reportIds, receiver, mockMetadata, mockActionHistory,
+            topic = Topic.FULL_ELR, format = Report.Format.FHIR
+        )
+
+        assertThat(report.bodyFormat).isEqualTo(Report.Format.FHIR)
+        assertThat(Regex("None-${report.id}-\\d*.fhir").matches(report.name)).isTrue()
+        assertThat(blobInfo.blobUrl).endsWith("/devstoreaccount1/container1/process%2Forg.name%2F${report.name}")
+    }
+
+    @Test
+    fun `test formExternalFilename from header when body url is not null`() {
+        val expectedReportId = UUID.randomUUID()
+        val expectedFileName = "sampleFileName-someReportId.fhir"
+        val expectedBodyUrl = "https://this/is/an/example/blob/url/$expectedFileName"
+
+        val reportUnderTest = ReportFile()
+        reportUnderTest.reportId = expectedReportId
+        reportUnderTest.bodyUrl = expectedBodyUrl
+
+        val header = WorkflowEngine.Header(
+            Task(),
+            reportUnderTest,
+            null,
+            null,
+            null,
+            null,
+            null,
+            true
+        )
+
+        assertThat(Report.formExternalFilename(header)).isEqualTo(expectedFileName)
+    }
+
+    @Test
+    fun `test formExternalFilename from header when body url is null`() {
```

Comment:
Maybe a tech debt ticket to figure out if this is ever true outside the covid pipeline?